src/store/store.js

```diff
@@ -0,0 +1,49 @@
+import Vue from 'vue';
+import Vuex from 'vuex';
+
+Vue.use(Vuex);
+
+export const store = new Vuex.Store({
+    state: {
+        cities: [ 
+            {id: 0, name: 'Satu Mare'},
+            {id: 1, name: 'Arad'},
+            {id: 2, name: 'Timisoara'},
+        ],
+    },
+    getters: {
+        getCities: state => {
+            return state.cities;
+        },
+        cityToString: (state) => (city) => {
+            return 'ID: ' + city.id + ' Name: ' + city.name;
+        },
+    },
+    mutations: {
+        addCity: (state, city) => {
+            //add new city to cities list
+            state.cities.push(city);
+        },
+        updateCity: (state, city) => {
+            //change city name
+            const selectedCity = state.cities.filter(item => item.id === city.id)[0];
+            selectedCity.name = city.name;
```

Comment:
Every time you update something in the store, you should always use a copy of the same object with the latest props :) 
Use the `Vue.set` instead :)